### PR TITLE
Full refactor and add RegularTask and PriorityTask classes

### PR DESCRIPTION
**Pull request by ruVlad**
Change WeeklyTask to abstract
Add RegularTask
Add PriorityClass
Add abstract method GetAlarm

### CODE REVIEW

**Comment by ruVlad** [comment]
Maybe the task with inheritance is not quite correct. But my priority value is always "Normal" as default, so I use only "PriorityTask" everywhere

**Comment by ruVlad** [comment]
I used interfaces in edit methods because I do not used class objects to filter methods

File: WeeklyTask/IPriorityTask.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace WeeklyTask
+{
+    interface IPriorityTask : IRegularTask
+    {
+        public PriorityEnum Priority { get; set; }
```
Comment (rubleuskia) [inline WeeklyTask/IPriorityTask.cs:5]: drop access modifier
```suggestion
        PriorityEnum Priority { get; set; }
```

File: WeeklyTask/Program.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace WeeklyTask
+{
+    class Program
+    {
+        private static readonly WeeklyTaskService s_service = new();
+
+        static void Main(string[] args)
+        {
+            s_service.RegisterWriteHandler(Write);
```
Comment (rubleuskia) [inline WeeklyTask/Program.cs:11]: 👍

**Review by rubleuskia — APPROVED** [review]
🚀

**Pull request merged**